gson/src/test/java/com/google/gson/internal/bind/Java17ReflectiveTypeAdapterFactoryTest.java

```diff
@@ -6,42 +6,39 @@
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.google.gson.TypeAdapter;
-import com.google.gson.internal.reflect.ReflectionHelperTest;
+import com.google.gson.internal.reflect.Java17ReflectionHelperTest;
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
 import java.io.IOException;
 import java.nio.file.attribute.GroupPrincipal;
 import java.nio.file.attribute.UserPrincipal;
 import java.security.Principal;
-import org.junit.AssumptionViolatedException;
 import org.junit.Before;
 import org.junit.Test;
 
-public class ReflectiveTypeAdapterFactoryTest {
+public class Java17ReflectiveTypeAdapterFactoryTest {
 
-  // The class jdk.net.UnixDomainPrincipal is one of the few Record types that are included in the
-  // JDK.
+  // The class jdk.net.UnixDomainPrincipal is one of the few Record types that are included in the JDK.
   // We use this to test serialization and deserialization of Record classes, so we do not need to
-  // have
-  // record support at the language level for these tests. This class was added in JDK 16.
+  // have record support at the language level for these tests. This class was added in JDK 16.
   Class<?> unixDomainPrincipalClass;
 
   @Before
   public void setUp() throws Exception {
-    try {
-      Class.forName("java.lang.Record");
-      unixDomainPrincipalClass = Class.forName("jdk.net.UnixDomainPrincipal");
-    } catch (ClassNotFoundException e) {
-      // Records not supported, ignore
-      throw new AssumptionViolatedException("java.lang.Record not supported");
-    }
+    unixDomainPrincipalClass = Class.forName("jdk.net.UnixDomainPrincipal");
+  }
+
+  // Class for which the normal reflection based adapter is used
+  private static class DummyClass {
+    @SuppressWarnings("unused")
+    public String s;
   }
 
   @Test
   public void testCustomAdapterForRecords() {
     Gson gson = new Gson();
     TypeAdapter<?> recordAdapter = gson.getAdapter(unixDomainPrincipalClass);
-    TypeAdapter<?> defaultReflectionAdapter = gson.getAdapter(UserPrincipal.class);
+    TypeAdapter<?> defaultReflectionAdapter = gson.getAdapter(DummyClass.class);
```

Comment:
Have changed this because `UserPrincipal` is an interface without any fields, so assuming that the normal reflection based adapter is used for it might be a bit brittle.